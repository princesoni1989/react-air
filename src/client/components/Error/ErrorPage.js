import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ErrorPage extends Component {
  static propTypes = {
    error: PropTypes.shape({
      name: PropTypes.string.isRequired,
      message: PropTypes.string.isRequired,
      stack: PropTypes.string.isRequired,
    }),
  };

  static defaultProps = {
    error: null,
  };

  render() {
    const { error } = this.props;
    if (__DEV__ && error) {
      return (
        <div>
          <h1>{error.name}</h1>
          <pre>{error.stack}</pre>
        </div>
      );
    }

    return (
      <div>
        <h1>Error</h1>
        <p>Sorry, a critical error occurred on this page.</p>
      </div>
    );
  }
}

export default ErrorPage;
