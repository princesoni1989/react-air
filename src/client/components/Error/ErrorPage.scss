html {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 2rem;
  height: 100%;
  font-family: sans-serif;
  text-align: center;
}

body {
  margin: 0;
}

.h1 {
  font-weight: 400;
  color: red;
}

pre {
  white-space: pre-wrap;
  text-align: left;
}
