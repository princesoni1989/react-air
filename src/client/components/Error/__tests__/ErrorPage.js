/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';

import ErrorPage from '../ErrorPage';

describe(ErrorPage.name, () => {
  it('should render correctly when error exists on dev environment', () => {
    const error = { name: 'someName', stack: 'someStack', message: 'someMessage' };
    expect(shallow(<ErrorPage error={ error }/>)).toMatchSnapshot();
  });

  it('should render correctly when there is no error on dev environment', () => {
    expect(shallow(<ErrorPage />)).toMatchSnapshot();
  });
});
