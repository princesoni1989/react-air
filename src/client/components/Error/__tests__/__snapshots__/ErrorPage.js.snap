// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should render correctly when error exists on dev environment 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage
    error={
      Object {
        "message": "someMessage",
        "name": "someName",
        "stack": "someStack",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          someName
        </h1>,
        <pre>
          someStack
        </pre>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "someName",
        },
        "ref": null,
        "rendered": "someName",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "someStack",
        },
        "ref": null,
        "rendered": "someStack",
        "type": "pre",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            someName
          </h1>,
          <pre>
            someStack
          </pre>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "someName",
          },
          "ref": null,
          "rendered": "someName",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "someStack",
          },
          "ref": null,
          "rendered": "someStack",
          "type": "pre",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ErrorPage should render correctly when there is no error on dev environment 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage
    error={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Error
        </h1>,
        <p>
          Sorry, a critical error occurred on this page.
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error",
        },
        "ref": null,
        "rendered": "Error",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry, a critical error occurred on this page.",
        },
        "ref": null,
        "rendered": "Sorry, a critical error occurred on this page.",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Error
          </h1>,
          <p>
            Sorry, a critical error occurred on this page.
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error",
          },
          "ref": null,
          "rendered": "Error",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry, a critical error occurred on this page.",
          },
          "ref": null,
          "rendered": "Sorry, a critical error occurred on this page.",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
